src/NLog/Config/LoggingConfigurationParser.cs

```diff
@@ -1029,24 +1028,15 @@ private void SetPropertyValueFromString(object targetObject, string propertyName
         {
             try
             {
-                var propertyValueExpanded = ExpandSimpleVariables(propertyValue, out var matchingLayout);
-                if (matchingLayout != null && propertyValueExpanded == propertyValue)
+                var propertyValueExpanded = ExpandSimpleVariables(propertyValue, out var matchingVariableName);
+                if (matchingVariableName != null && propertyValueExpanded == propertyValue && TryLookupDynamicVariable(matchingVariableName, out var matchingLayout) && PropertyHelper.TryGetPropertyInfo(targetObject, propertyName, out var propInfo) && propInfo.PropertyType.IsAssignableFrom(matchingLayout.GetType()))
```

Comment:
Maybe exact condition to method for readability?

Reply:
Will extract into helper-method.